### ISSUE TITLE
Writing frame immediately after window creation doesn't work?

### ISSUE THREAD

**Issue by josephrocca** [emoon/rust_minifb#81]
Sorry if I've missed something in the docs about this. The following example produces a window filled with black pixels even though I'm passing it white pixels:

```
use minifb::{WindowOptions, Window};

fn main() {

    let input_size: usize = 64;
    let mut input_buffer = vec![16777215; input_size.pow(2)]; // white buffer
    let mut window = Window::new("Input", input_size, input_size, WindowOptions::default()).unwrap();
    window.update_with_buffer(&input_buffer).unwrap();
    std::thread::sleep(std::time::Duration::from_millis(10000));

}
```

If I add a 100ms sleep right after creating the window (before the `update_with_buffer` line), then it works fine. So it seems like the window needs some "setup time"?

**Comment by emoon** [comment]
I'm not sure what the problem is here but it's likely related to something in the OS. Minifb mostly calls to the OS level functions (such as `StretchBlit` on Windows) I'm not sure if the window needs to be updated a couple of time before it takes effect.

**Comment by josephrocca** [comment]
For the record, I'm running Ubuntu 18.04. I don't have access to windows or mac to see if it's just isolated to Ubuntu/Linux. It's easy to work around anyway so it's not a big problem. By the way - while I've got you: thanks so much for the very handy crate! It's probably about third in my most-used crates after rand and rayon.

**Comment by emoon** [comment]
Yeah. My point was just that it likely differs depending on how each operating system handles updates to the window so it's hard to say a strict guideline here. Personally I have never used minifb for doing just one update but have a loop that shows something until end. Usually you have something that is outlined in the example such as

```Rust
  while window.is_open() {
    window.update_with_buffer(...);
  }
```

It might be that the update needs to be called a bunch of times.

Also thanks! It's great to see that people find it useful :) Makes the work worth it!
